packages/ui-components/src/components/DateRangePicker/WeekPicker.tsx

```diff
@@ -24,10 +24,14 @@ export const WeekPicker = ({
   maxMomentDate,
   weekDisplayFormat,
   onChange,
-}) => {
-  const { pickerFormat, modifier } = weekDisplayFormat
+}: WeekPickerProps) => {
+  // need to cast string of modifier to type
+  const { pickerFormat, modifier } = (weekDisplayFormat
     ? WEEK_DISPLAY_CONFIG[weekDisplayFormat]
-    : GRANULARITY_CONFIG[GRANULARITIES.WEEK];
+    : GRANULARITY_CONFIG[GRANULARITIES.WEEK]) as {
+    pickerFormat: string;
+    modifier?: ModifierType;
```

Comment:
I really like how you've handled this tricky case. I think casting makes sense and it's also nice to have a ModifierType to use when needed.